### PR TITLE
Bug 2034484: feat: library-go bump

### PR DESCRIPTION
**Pull request by qJkee**
Bumping the library-go dependency to latest, this should take advantage of leader election changes for SNO clusters proposed in this [library-go PR](https://github.com/openshift/library-go/pull/1242) and performance improvements in this [PR](https://github.com/openshift/library-go/pull/1235).

It also includes the following BZ bug fix 
Bug 2034484: fix:library-go nil pointer issue

### CODE REVIEW

**Comment by stlaz** [comment]
Please move any changes done outside the vendoring to a separate commit.

File: bindata/v4.0.0/controller/clusterrole.yaml [hunk 1]
```diff
@@ -62,3 +62,11 @@ rules:
...
   - watch
   - update
+- apiGroups:
+  - config.openshift.io
+  resources:
+  - infrastructures
+  verbs:
+  - get
+  - list
+  - watch
```
Comment (stlaz) [inline bindata/v4.0.0/controller/clusterrole.yaml:72]: separate commit, explain in commit message please

File: cmd/service-ca-operator/main.go [hunk 2]
```diff
@@ -23,6 +23,7 @@ func main() {
 	pflag.CommandLine.SetNormalizeFunc(utilflag.WordSepNormalizeFunc)
 	pflag.CommandLine.AddGoFlagSet(goflag.CommandLine)
 
+	logs.AddFlags(pflag.CommandLine)
```
Comment (stlaz) [inline cmd/service-ca-operator/main.go:26]: separate commit, fix like this: https://github.com/openshift/cluster-authentication-operator/commit/a2f0d1f994c7d8a8abe5c9beef4b97e02e244608

File: pkg/operator/operator.go [hunk 3]
```diff
@@ -106,7 +106,7 @@ func (c *serviceCAOperator) Sync(ctx context.Context, syncCtx factory.SyncContex
 }
 
 func (c serviceCAOperator) updateStatus(operatorConfig *operatorv1.ServiceCA) {
-	v1helpers.UpdateStatus(c.operatorClient, func(status *operatorv1.OperatorStatus) error {
+	v1helpers.UpdateStatus(context.TODO(), c.operatorClient, func(status *operatorv1.OperatorStatus) error {
```
Comment (stlaz) [inline pkg/operator/operator.go:109]: wire the context properly

**Review by stlaz — CHANGES_REQUESTED** [review]
please separate the changes to separate commits and explain them

File: pkg/operator/operator.go (hunk 3 above)
Comment (qJkee) [reply pkg/operator/operator.go]: bb03417

File: cmd/service-ca-operator/main.go (hunk 2 above)
Comment (qJkee) [reply cmd/service-ca-operator/main.go]: 85ccd9a

File: bindata/v4.0.0/controller/clusterrole.yaml (hunk 1 above)
Comment (qJkee) [reply bindata/v4.0.0/controller/clusterrole.yaml:72]: 89f04d8

**Review by stlaz — APPROVED** [review]

**Comment by stlaz** [comment]
/lgtm
thank you for your PR :+1:

**Pull request merged**